### DISCUSSION TITLE
Sistema para organização de meetups

### DISCUSSION THREAD

**Discussion by itepifanio** [apyb/comunidade#81] [Geral]
Algo que todo mundo que organiza meetups tem problema é com controle de check-in/check-out e emissão de certificados para os participantes/palestrantes (com código único e validador de certificado).

No Grupy RN (PUG do Rio Grande do Norte) utilizamos [o Jararaca](https://github.com/grupyrn/jararaca), um software desenvolvido pela comunidade que: aceita cadastro no evento, permite controle de checkin e checkout dos participantes e emite certificado simples (ainda não temos um código único/validador de certificado). 

Acredito que várias comunidades utilizem uma forma própria de resolver esse problema de geração de certificado, seja utilizando scripts [[1]](https://github.com/gabubellon/python-brasil-certificate-generator), [[2]](https://github.com/sedir/certificados), sistemas pagos ou sistemas gratuitos mas limitados. Entretanto, acredito que fosse interessante uma iniciativa da APyB para criação de um sistema que fosse comum a todas as comunidades descentralizadas, isso ajudaria muito inclusive a verificar a atividade das comunidades no país.

Edit:
Esse sistema evitaria organizadores de manualmente submeterem seus eventos a listas como [a de divulgação de eventos](https://github.com/apyb/comunidade/discussions/80) ou [abrir PRs em repositórios descentralizados da APyB](https://github.com/rougeth/pythonnobrasil#-eventos-python-no-brasil).

**Comment by rougeth** [comment]
Oi @itepifanio! Eu tive poucos problemas com checkin e checkout de participantes nos eventos que organizei, mas emissão de certificados ainda é um problema recorrente. Nos últimos anos que participei da organização da Python Brasil, esse tópico sempre demandou muito tempo e esforço.

O que falta para vocês do Grupy RN usarem as soluções que outros eventos criaram ou usaram? Caso elas fossem feitas pela APyB, qual diferença faria? Ou o que você espera que seria diferente?

Pergunto isso porque uma iniciativa vinda da APyB não teria muita diferença de uma iniciativa do Grupy RN, por exemplo. A APyB é só mais uma comunidade Python no Brasil.

**Reply by itepifanio** [reply]
> O que falta para vocês do Grupy RN usarem as soluções que outros eventos criaram ou usaram?

Acho que mais uma catalogação desse ferramental. Vejo a galera utilizando o meetup.com, mas ele não tem emissão de certificado, nem controle de checkin, checkout.

> Caso elas fossem feitas pela APyB, qual diferença faria? Ou o que você espera que seria diferente? [...] A APyB é só mais uma comunidade Python no Brasil

A APyB é o mais próximo de centralizar as demais comunidades e a adoção de uma solução open source permitiria que mais pessoas pudeseem contribuir para a ferramenta, além de diminuir os custos por dividir entre várias comunidades. O que seria diferente é o desenvolvimento mais horizontal da ferramenta, além de manutenção constante por termos várias pessoas interessadas no sistema.

Eu vi que a PSF (Python Software Foundation) utiliza o meetup.com e estamos investigando a possibilidade de um checkin/checkout que se comunique com a plataforma e emita certificados.

**Reply by rougeth** [reply]
> Acho que mais uma catalogação desse ferramental.

Isso seria ótimo mesmo.

**Comment by jjpaulo2** [comment]
Oi @itepifanio. Acho legal a ideia de ter algo centralizado, mas concordo com o @rougeth. Acho que isso rende uma discussão muito boa, mas não como uma responsabilidade da APyB. 

Aqui em Blumenau também temos problemas com check-in e emissão de certificados nos nossos encontros do GruPy e hoje utilizamos o [meetup.com](https://meetup.com) pro check-in e um [projeto do hackerspace](https://github.com/HackerspaceBlumenau/certificados-hs) pra emissão dos certificados.

Há alguns anos, era muito usado o [Speakerfight](https://speakerfight.com/) pra fazer votação de palestras, mas o sistema tá um pouco abandonado e tem vários problemas. Tentei até entrar em contato com o Luan, mas sem sucesso, pra saber se tinha como ajudar a manter ele. @itepifanio talvez fosse o caso de pegar um desses projetos mais "maduros", como o Speakerfight ou até mesmo o Jararaca de vocês ai no RN, e entender como evoluir e transformar isso em um padrão das comunidades. Podemos inclusive criar uma comunidade em torno de manter essa plataforma.

Fico a disposição pra ajudar a construir um projeto desses.

**Reply by itepifanio** [reply]
> mas não como uma responsabilidade da APyB [...] criar uma comunidade em torno de manter essa plataforma

@jjpaulo2 acho que a ideia seria essa mesmo, integrar o ferramental usado pelas comunidades pra gerar uma forma mais fácil e talvez escrever [um guia](https://apyb.python.org.br/guias/como-organizar-eventos/) de como utilizar esse ferramental pra facilitar a organização dos meetups.

Minha ideia seria utilizar o github da APyB porque centraliza mais a adoção do ferramental e criar um time pra desenvolver essa plataforma, não precisaria ser responsabilidade da APyB e idealmente teriam responsáveis de várias comunidades

**Reply by rougeth** [reply]
> Minha ideia seria utilizar o github da APyB porque centraliza mais a adoção do ferramental e criar um time pra desenvolver essa plataforma, não precisaria ser responsabilidade da APyB e idealmente teriam responsáveis de várias comunidades

Usar o Github da APyB para projetos da comunidade não é sinônimo de engajamento, infelizmente. O que acontece depois de um ano ou dois (ou 7 - caso recente) que o projeto fica parado e quem era o responsável na época não está mais presente? Sendo um projeto dá APyB (se está no /apyb, é de responsabilidade da APyB), é esperado que a associação mantenha aquele projeto e, pelo menos nos últimos anos, a direção da APyB não tem condições de manter projetos de código aberto além dos que já tem.

Eu sou muito a favor de uma iniciativa da comunidade para esse problema em específico, eu particularmente tive muitos problemas com a parte de certificação, e seria uma dor de cabeça a menos na Python Brasil.

Então minha sugestão é você @itepifanio puxar esse projeto. Eu lembro de alguns problemas que tivemos nos últimos anos, e posso fazer a ponte com as pessoas que estão organizando os eventos esse ano.

**Reply by itepifanio** [reply]
> Usar o Github da APyB para projetos da comunidade não é sinônimo de engajamento, infelizmente

@rougeth meu pressuposto era esse :/

> Então minha sugestão é você @itepifanio puxar esse projeto

Vou conseguir me dedicar a uma aplicação do tipo a partir de outubro. Acredito que consiga conversar com alguns organizadores de eventos pra entender os requisitos pra um MVP de geração de certificado pras comunidades.
